### DISCUSSION TITLE
Exporting plot data to a text file

### DISCUSSION THREAD

**Discussion by Ibrahim-Ajibosin** [pyvista/pyvista#2274] [Q&A] [answered]
I made a plot from my model data and from my mesh object i am trying to  export the model data in the array to a text file but the problem is i only want the data used in making the plot not all the model data. Below is the plot and also a screenshot of my code 
<img width="1280" alt="Screenshot 2022-02-25 at 12 54 35" src="https://user-images.githubusercontent.com/82901442/155712948-fb3a240f-107f-442e-9489-256b3e21b8f1.png">
![Figure_1](https://user-images.githubusercontent.com/82901442/155713219-f6e12b51-47f5-42bd-8ef1-513431c38502.png)

**Comment by RichardScottOZ** [comment, marked as answer]
You want sample_over_line, which is what plot over line is using...that will return the result you can then save the data from

**Reply by Ibrahim-Ajibosin** [reply]
> You want sample_over_line, which is what plot over line is using...that will return the result you can then save the data from

i tried the sample_over_line function but my array 1 and 2 are still of the size 123370 which is still all the data points from my model, if you can help with a sample code of how i can get just the sampled points from the plot  i would be grateful

**Reply by MatthewFlamm** [reply]
I can't read your code very well from the screenshot, next time it would be useful to have the code pasted directly. You need to do something like:

```py
mesh = pv.read(....)
sampled_line = mesh.sample_over_line(point1, point2, resolution)
array_1 = sampled_line['array_1']
```

**Comment by RichardScottOZ** [comment]
https://docs.pyvista.org/api/core/_autosummary/pyvista.StructuredGrid.sample_over_line.html

**Comment by RichardScottOZ** [comment]
Your mesh will still be the same - but the 'sample' should be different?

**Reply by Ibrahim-Ajibosin** [reply]
Thanks richard the sample_over_line function worked but i had to set the resolution to get the specified amount of data set i want

**Reply by RichardScottOZ** [reply]
Ok, great.

**Reply by Ibrahim-Ajibosin** [reply]
Thanks, i will like to ask another question, my code is for one location, how can i export data for several location at once , i am thinking maybe i can use a list of list, what do you think?

**Reply by RichardScottOZ** [reply]
You mean many lines at once?

**Reply by Ibrahim-Ajibosin** [reply]
Exactly more profiles at once and then i export them into files

**Comment by RichardScottOZ** [comment]
![image](https://user-images.githubusercontent.com/72196131/155809714-e88f77f9-2e7f-4074-9ccb-ac7f009be0ec.png)
